### PR TITLE
MRG: Make get_head_mri_trans more permissive re landmark names

### PR DESCRIPTION
**Pull request by hoechenberger**
Closes #769

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [ ] All CIs are happy
- [ ] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/main/doc/whats_new.rst) is updated
- [x] PR description includes phrase "closes <#issue-number>"

### CODE REVIEW

File: mne_bids/read.py [hunk 1]
```diff
@@ -637,16 +637,31 @@ def get_head_mri_trans(bids_path, extra_params=None):
     with open(t1w_json_path, 'r', encoding='utf-8-sig') as f:
         t1w_json = json.load(f)
     mri_coords_dict = t1w_json.get('AnatomicalLandmarkCoordinates', dict())
-    mri_landmarks = np.asarray((mri_coords_dict.get('LPA', np.nan),
-                                mri_coords_dict.get('NAS', np.nan),
-                                mri_coords_dict.get('RPA', np.nan)))
+
+    # landmarks array: rows: [LPA, NAS, RPA]; columns: [x, y, z]
+    mri_landmarks = np.empty((3, 3))
+    mri_landmarks[:] = np.nan
```
Comment (agramfort) [inline mne_bids/read.py:643]:
```suggestion
    mri_landmarks = np.full((3, 3), np.nan)
```

File: mne_bids/tests/test_read.py [hunk 2]
```diff
@@ -238,17 +238,40 @@ def test_get_head_mri_trans(tmpdir):
     assert trans['from'] == estimated_trans['from']
     assert trans['to'] == estimated_trans['to']
     assert_almost_equal(trans['trans'], estimated_trans['trans'])
-    print(trans)
-    print(estimated_trans)
 
     # provoke an error by introducing NaNs into MEG coords
+    raw.info['dig'][0]['r'] = np.ones(3) * np.nan
```
Comment (agramfort) [inline mne_bids/tests/test_read.py:243]:
```suggestion
    raw.info['dig'][0]['r'] = np.full(3, np.nan)
```

File: mne_bids/tests/test_read.py [hunk 3]
```diff
@@ -238,17 +238,40 @@ def test_get_head_mri_trans(tmpdir):
...
-        raw.info['dig'][0]['r'] = np.ones(3) * np.nan
-        sh.rmtree(anat_dir)
-        bids_path = write_anat(t1w_mgh, bids_path=t1w_bidspath,
-                               raw=raw, trans=trans, verbose=True)
         estimated_trans = get_head_mri_trans(bids_path=bids_path)
 
+    # test we are permissive for different casings of landmark names in the
+    # sidecar, and also accept "nasion" instead of just "NAS"
+    print('--------->>>>>>')
+    print(raw_fname)
```
Comment (agramfort) [inline mne_bids/tests/test_read.py:253]: cleanup?

File: mne_bids/read.py (hunk 1 above)
Comment (hoechenberger) [reply mne_bids/read.py:643]: Nice trick, I didn't know about this! 👌

**Comment by hoechenberger** [comment]
@agramfort All addressed!

**Review by agramfort — APPROVED** [review]

**Comment by agramfort** [comment]
@sappelhoff I let you click the green button !

**Review by sappelhoff — APPROVED** [review]

**Comment by sappelhoff** [comment]
Thanks @hoechenberger

**Pull request merged**